### PR TITLE
tileGrid: dynamically render tiles based on an index.yaml

### PR DESCRIPTION
**Pull request by tomleah**
## Checklist

- [x] commit message follows [commit guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).
- [x] website builds and runs in production - for information on how to test locally go [here](https://github.com/appsody/website/blob/master/DEVELOPMENT.md).

## Description

- Added a sample index to be used in developement
- Pulls down indexes from a url to use with rendering tiles
- Updated Travis to remove development indexes before deploying
## Related Issues
fixes: https://github.com/appsody/website/issues/51

### CODE REVIEW

File: ci/get-indexes.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# !/bin/sh
+# Remove all development indexes
+rm -rf src/data/indexes/*
+
+# Download indexes that will be used on production (appsody.dev)
+cd src/data/indexes
+# Repeat the following command to add new indexes.
+# Incubator
+wget https://github.com/appsody/stacks/releases/download/java-microprofile-v0.2.4/incubator-index.yaml
```
Comment (neeraj-laad) [inline ci/get-indexes.sh]: I think this URL should be the latest index?

Comment (neeraj-laad) [reply ci/get-indexes.sh]: perhaps we can add all 3 indexes here now, even if stable is blank?

File: gatsby-node.js [hunk 2]
```diff
@@ -1,4 +1,24 @@
 const path = require(`path`)
+const fetch = require('node-fetch');
+const fs = require('file-system');
+
+// Add URLs to v2 indexes to this array for them to be rendered on the website
+// If the index does not need to be downloaded, place the index.yaml in src/data/indexes
+// Note: this is only for developement add URLs to /ci/get-indexes.sh to affect production
+const indexURLs = [
+  'https://github.com/appsody/stacks/releases/latest/download/incubator-index.yaml'
```
Comment (neeraj-laad) [inline gatsby-node.js]: again perhaps we can add all 3 right away?

File: ci/get-indexes.sh (hunk 1 above)
Comment (tomleah) [reply ci/get-indexes.sh]: Yes this should actually be `https://github.com/appsody/stacks/releases/latest/download/incubator-index.yaml`

Comment (tomleah) [reply ci/get-indexes.sh]: Will add these, we might want to make sure we land https://github.com/appsody/website/issues/83 to be able to tell apart the repositories

File: gatsby-node.js (hunk 2 above)
Comment (tomleah) [reply gatsby-node.js]: Added 🙂

File: src/components/tileGrid.js [hunk 3]
```diff
@@ -4,13 +4,17 @@ import Tile from "../components/tile";
...
     
+    const defaultRepo = "incubator"
     const tiles = props.stacks.map(stack => {
         if (stack == null) return null;
         
         const templateURL = stack.templates[0].url;
         const repo = templateURL.split("/").reverse()[0].split(".")[0];
         const githubURL = `https://github.com/appsody/stacks/tree/master/${repo}/${stack.id}`;
 
+        if (!stack.templates[0].url.includes(defaultRepo)) {
```
Comment (neeraj-laad) [inline src/components/tileGrid.js]: could we use a more specific pattern? `/defaultRepo.`

File: src/components/tileGrid.js [hunk 4]
```diff
@@ -4,13 +4,17 @@ import Tile from "../components/tile";
...
+            return <Tile id={stack.id} heading={stack.name} desc={stack.description} cmd={"appsody init " + repo+"/"+stack.id} github={githubURL}/>
+        }
```
Comment (neeraj-laad) [inline src/components/tileGrid.js]: I personally find it confusing to have a return within an if and then a return when the if fails.. Could we use a if/else to set the value for tile and then return the tile?

Comment (mhamwala) [reply src/components/tileGrid.js]: change addressed

**Pull request merged**